R/error-checks.R

```diff
@@ -69,6 +69,22 @@ validateIsNumeric <- function(object, nullAllowed = FALSE) {
   validateIsOfType(object, c("numeric", "integer"), nullAllowed)
 }
 
+validateIsInteger <- function(object, nullAllowed = FALSE) {
+  if (nullAllowed && is.null(object)) {
+    return()
+  }
+
+  if (all(floor(object) == object, na.rm = TRUE)) {

```

Comment:
Very weird way to check for integer. Does not work with the integer keywords otherwise as expected